src/control/cmd/agent/main.go

```diff
@@ -57,17 +57,69 @@ type cliOptions struct {
 	Insecure   bool       `short:"i" long:"insecure" description:"have agent attempt to connect without certificates"`
 	RuntimeDir string     `short:"s" long:"runtime_dir" description:"Path to agent communications socket"`
 	LogFile    string     `short:"l" long:"logfile" description:"Full path and filename for daos agent log file"`
+	NetworkDbg bool       `short:"n" long:"netdebug" description:"Enable extended network device scan debug output"`
 	Version    versionCmd `command:"version" description:"Print daos_agent version"`
+	NetScan    netScanCmd `command:"scan" description:"Perform local fabric scan"`
 }
 
 type versionCmd struct{}
+type netScanCmd struct {
+	Debug          bool   `short:"d" long:"debug" description:"Enable debug output"`
```

Comment:
Doesn't this conflict with the top-level debug? It's a little confusing either way.

Reply:
Currently, it does not conflict because the top level options aren't passed in and don't appear to be globally available.  Should I be setting a custom implementation of p.CommandHandler = func(cmd flags.Commander, args []string) error { ... } to call the execute function with the cliOptions, or is there some more reasonable way to do it?  I thought the proposal was relatively simple and non-invasive, but am perfectly onboard with doing it differently.

Reply:
I've made updates to address this.  Now correctly uses the top level option.